src/tracer.js

```diff
@@ -185,6 +185,11 @@ export default class Tracer {
         }
 
         let parent: ?SpanContext = options.childOf instanceof Span ? options.childOf.context(): options.childOf;
+
+        if (!parent && options.references.length > 0) {
+            parent = options.references[0].referencedContext();
+        }
```

Comment:
this is already (mostly) taken care of by L198, and doing this unconditionally here is wrong anyway, because the ref type can be something that should not become a parent.